google-cloud-storage/src/test/java/com/google/cloud/storage/GapicUnbufferedWritableByteChannelTest.java

```diff
@@ -176,16 +192,113 @@ public void resumableUpload() throws IOException, InterruptedException, Executio
     }
   }
 
+  @Test
+  public void resumableUpload_chunkAutomaticRetry()
+      throws IOException, InterruptedException, ExecutionException {
+    AtomicBoolean req2SendErr = new AtomicBoolean(true);
+    AtomicBoolean req4SendErr = new AtomicBoolean(true);
+    AtomicInteger writeCount = new AtomicInteger(0);
+    StorageImplBase service =
+        new DirectWriteService(
+            (obs, requests) -> {
+              writeCount.getAndIncrement();
+              if (requests.equals(ImmutableList.of(req1))) {
+                obs.onNext(resp1);
+                obs.onCompleted();
+              } else if (requests.equals(ImmutableList.of(req2))) {
+                obs.onNext(resp2);
+                if (req2SendErr.get()) {
+                  req2SendErr.set(false);
+                  obs.onError(apiException(Code.DATA_LOSS));
```

Comment:
Are you going to test other retriable gRPC codes?

Reply:
This is specifically a non-default retryable status code to ensure our retry wiring is working.

The other grpc status codes that we do want to retry are covered via tests for the translation from grpc to StorageException.